pkg/odo/cli/service/backends.go

```diff
@@ -0,0 +1,422 @@
+/*
+	This file contains code for various service backends supported by odo. Different backends have different logics for
+	Complete, Validate and Run functions. These are covered in this file.
+*/
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	scv1beta1 "github.com/kubernetes-sigs/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/odo/cli/service/ui"
+	commonui "github.com/openshift/odo/pkg/odo/cli/ui"
+	svc "github.com/openshift/odo/pkg/service"
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/klog"
+)
+
+// This CompleteServiceCreate contains logic to complete the "odo service create" call for the case of Operator backend
+func (b *OperatorBackend) CompleteServiceCreate(o *CreateOptions, cmd *cobra.Command, args []string) (err error) {
+	// since interactive mode is not supported for Operators yet, set it to false
+	o.interactive = false
+
+	// if user has just used "odo service create", simply return
+	if o.fromFile == "" && len(args) == 0 {
+		return
+	}
+
+	// if user wants to create service from file and use a name given on CLI
+	if o.fromFile != "" {
+		if len(args) == 1 {
+			o.ServiceName = args[0]
+		}
+		return
+	}
+
+	// split the name provided on CLI and populate servicetype & customresource
+	o.ServiceType, b.CustomResource, err = svc.SplitServiceKindName(args[0])
+	if err != nil {
+		return fmt.Errorf("invalid service name, use the format <operator-type>/<crd-name>")
+	}
+
+	// if two args are given, first is service type and second one is service name
+	if len(args) == 2 {
+		o.ServiceName = args[1]
+	}
+
+	return nil
+}
+
+func (b *OperatorBackend) ValidateServiceCreate(o *CreateOptions) (err error) {
+	d := NewDynamicCRD()
```

Comment:
Since the `DynamicCRD` struct and it's functions are specific to operator hub, please move them out of the create.go

Reply:
Moved to `operator_backend.go`.